### PR TITLE
Use RPC tag for all artifact publishing and consuming

### PR DESCRIPTION
**Pull request by odyssey4me**
The implementation to discover the current release tag has been implemented in the RPC group_vars to replace the current module-based method. This mechanism is far simpler and does not rely on task execution to derive it.

In order to override the openstack_release variable to match the value of rpc_release, the rpc_release value must be available without requiring the use of a module and task. The values are therefore set in the user_*.yml override files.

Connected https://github.com/rcbops/u-suk-dev/issues/765

### CODE REVIEW

**Comment by rackernexus** [comment]
Your content preview [is now ready](https://staging.developer.rackspace.com/staging.horse/build-ddbb54cd35/docs/private-cloud/release-notes/). :bow:

File: rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml [hunk 1]
```diff
@@ -164,3 +164,7 @@ glance_flavor: keystone
 glance_glance_api_conf_overrides:
   DEFAULT:
     image_cache_dir: None
+
+# The OSA release tag is overridden so that all artifacts
+# produced and consumed use the RPC version tag.
+openstack_release: "lookup('pipe', 'cd /opt/rpc-openstack && git describe --tags --abbrev=0')"
```
Comment (evrardjp) [inline rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml]: {{ }} missing

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/staging.horse/build-407409155d/docs/private-cloud/release-notes/)

File: rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml (hunk 1 above)
Comment (odyssey4me) [inline rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml]: @evrardjp Done, thanks for picking that up!

**Comment by rackernexus** [comment]
:eyes: [Preview](https://staging.developer.rackspace.com/staging.horse/build-f17bc9cbb1/docs/private-cloud/release-notes/)

**Comment by rackernexus** [comment]
HULK [PREVIEW](https://staging.developer.rackspace.com/staging.horse/build-c1c31b7e9a/docs/private-cloud/release-notes/)

File: rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml [hunk 2]
```diff
@@ -185,3 +185,7 @@ glance_flavor: keystone
 glance_glance_api_conf_overrides:
   DEFAULT:
     image_cache_dir: None
+
+# The OSA release tag is overridden so that all artifacts
+# produced and consumed use the RPC version tag.
+openstack_release: "{{ lookup('pipe', 'cd /opt/rpc-openstack && git describe --tags --abbrev=0') }}"
```
Comment (hughsaunders) [inline rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml]: Could this be set to rpc_release to make it explicit that they are being set to the same value.

Comment (odyssey4me) [inline rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml]: The trouble is that ``rpc_release`` doesn't exist in the global scope. I'd have to have the var set in over of the user_*.yml files. Do you think this is a suitable solution?

**Comment by odyssey4me** [comment]
recheck_ceph
recheck_swift

**Comment by evrardjp** [comment]
squash and we are good to merge?

**Comment by odyssey4me** [comment]
I don't believe it should be squashed, personally.

**Comment by evrardjp** [comment]
Oh no, indeed.

**Pull request merged**